src/v2g_exi_codec_gen/base_coder_classes.py

```diff
@@ -375,9 +375,9 @@ def _debug_element_particle_properties(element: ElementData):
         grammar = self.create_empty_grammar()
 
         def _add_subsequent_grammar_details(element: ElementData, particle: Particle,
-                                            grammar: ElementGrammar,
                                             particle_index: int, index_last_nonoptional_particle: int,
                                             particle_is_part_of_sequence: bool):
+            nonlocal grammar
```

Comment:
`nonlocal` often is not a good idea

Reply:
(Python docs tell me that as it is a class, it should be accessible anyway, but it isn't, so d'uh.)

I operated this inner function without `nonlocal` before, and actually passed in the grammar. The issue which occurred: Inside the inner function, the `grammar` can be pushed to a list and a new one created. This now object connected to the same variable did not propagate to the variable passed in. (Not sure if it should have - the object replacement thing is the only explanation I have.) Hence when `grammar` was again accessed outside the inner function, it was modifying the original object (which was already stacked to said list), and thereby ruining it.